addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/device/cosem/CosemHexString.java

```diff
@@ -31,15 +33,19 @@
      */
     @Override
     protected StringType getStateValue(String cosemValue) throws ParseException {
-        String cosemHexValue = cosemValue.replaceAll("\\r\\n", "");
+        final String cosemHexValue = cosemValue.replaceAll("\\r\\n", "");
```

Comment:
Is it really needed, I don't like it for local variables and its not really the style of the rest of the binding(s) as well

Reply:
See this discussion: https://github.com/eclipse/smarthome/pull/6427#discussion_r229631533 This was for me the cue to actual use final on local variables (as I'm used to do). There is no special rule to use it or not, but a personal preference. Also it's not suggested to go in and change all locals to final just because. Therefore if you don't mind to much I prefer to use it this way.

Reply:
Afaik this does not match the result of the last chats I had about this with @kaikreuzer. So maybe he can share his current point of view.